### ISSUE TITLE
dropOnDelete Azure database for PostgreSQL

### ISSUE THREAD

**Issue by pietervincken** [movetokube/postgres-operator#52]
Hi

When using the operator to manage an Azure Database for PostgreSQL, the operator fails when the dropOnDelete option is enabled on the databases. 

Configuration used:
```
apiVersion: db.movetokube.com/v1alpha1
kind: Postgres
metadata:
  name: test-db
  namespace: dev-0
spec:
  database: test-db
  dropOnDelete: true
  schemas:
  - test_db
```

When this object is deleted from the Kubernetes API the following error appears:
```
{"level":"info","ts":1601564677.148541,"logger":"controller_postgres","msg":"Reconciling Postgres","Request.Namespace":"sit-1","Request.Name":"test-db"}
 {"level":"info","ts":1601564677.3637729,"logger":"controller_postgres","msg":"connected to postgres server","Request.Namespace":"sit-1","Request.Name":"test-db"}
 {"level":"error","ts":1601564677.3776052,"logger":"kubebuilder.controller","msg":"Reconciler error","controller":"postgres-controller","request":"sit-1/test-db","error":"pq: role \"test-db-group
 \" cannot be dropped because some objects depend on it","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\nsigs.k8s.io/controller-runtime/p
 kg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.1.10/pkg/internal/controller/controller.go:217\nsigs.k8s.io/controller-runtime/pkg/internal/contro
 ller.(*Controller).Start.func1\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.1.10/pkg/internal/controller/controller.go:158\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/pkg/mod/k8s.io/ap
 imachinery@v0.0.0-20190221213512-86fb29eff628/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190221213512-86fb29eff628/pkg/util/wait/w
 ait.go:134\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190221213512-86fb29eff628/pkg/util/wait/wait.go:88"}
```

We're running version 0.4.1 of the operator. We're running the operator for containerized databases as well (Zalando Postgres Operator) and there we don't see the same issue. 

Upon looking into the issue, I noticed that for the containerized databases, the ownership was properly transferred before the roles and database were deleted. Unfortunately, this seems to be the problem area on Azure. The re-assigning of the ownership fails on Azure due to the usage of the standard username instead of the one where the @ is stripped. 

This could be solved by overloading the getUser function in the Azure class and re-using the getLogin I think? (Not really a go developer :( )

**Comment by hitman99** [comment]
Thanks for opening the issue. It seems like a minor fix as there is already a mechanism which allows to do different things based on the `POSTGRES_CLOUD_PROVIDER` environment variable. I'll look into this this weekend probably, when I have more time.

**Comment by hitman99** [comment]
This was resolved in [v0.4.4](https://github.com/movetokube/postgres-operator/releases/tag/0.4.4), closing issue
